plugin.image.daily_dilbert/addon.py

```diff
@@ -0,0 +1,320 @@
+def build_url(query):
+	return g_Args_URL + '?' + urllib.urlencode(query)
+	
+def add_directory(mode=None,year='',month='',day='', page='',name='',icon='',fanart=''):
+	url = build_url({'mode': mode, 'foldername': name, 'page': page, 'year': year, 'month': month, 'day': day})
+	li = xbmcgui.ListItem(name, iconImage=icon)
+	xbmcplugin.addDirectoryItem(handle=g_AddonHandle, url=url, listitem=li, isFolder=True)
+
+def read_cache(date):
+	# Reads a cached scraped URL if existent
+	filename=date.strftime('%Y-%m-%d') + '.link'
+	return get_cacheddata(filename)
+
+def write_cache(date,url):
+	# Writes a scraped URL to the cache directory
+	filename=date.strftime('%Y-%m-%d') + '.link'
+	return set_cacheddata(url,filename)
+
+def get_cacheddata(filename):
+	# Reads a cached data from file if existent
+	# filename: lastmode/lastpage
+	cache_file=os.path.join(g_CacheDir + filename)
+	if os.path.isfile(cache_file):
+		try:
+			text_file = open(cache_file, "r")
+			data=text_file.read()
+			text_file.close()
+		except:
+			return ""
+		return data
+	else:
+		return ""
+		
+def set_cacheddata(param,filename):
+	# Writes a data to the cache directory
+	# filename: lastmode(lastpage)
+	cache_file=os.path.join(g_CacheDir + filename)
+	try:
+		text_file = open(cache_file, "w")
+		text_file.write(param)
+		text_file.close()
+		return True	
+	except:
+		return False
+
+def delete_cachefile(filename):
+	file=os.path.join(g_CacheDir + filename)
+	if os.path.isfile(file):
+		try:
+			os.remove(file)
+			return True
+		except:
+			return False
+	else:
+		return False
+		
+def get_image_url(date):
+	# Scrapes Dilbert website for URL to get according Dilbert comic strip from given date.
+	# Can be used with or without caching function (controlled by g_UseCache).
+	url = read_cache(date) if g_UseCache else "";
+	if url:
+		return url
+	else:
+		req=urllib2.Request(g_BaseUrl+date.strftime("%Y-%m-%d")+'/')
+		req.add_header('User-Agent', ' Mozilla/5.0 (Windows; U; Windows NT 5.1; en-GB; rv:1.9.0.3) Gecko/2008092417 Firefox/3.0.3')
+		try:
+			response=urllib2.urlopen(req)
+			page=response.read()
+			response.close()
+			del response
+		except:
+			msg=[xbmcaddon.Addon().getname() + ''': Couldn't scrape Dilbert webpage for the date '''+str(date.year)+'/'+str(date.month)+'/'+str(date.day)+'''. Please check your internet connection.''','''To be sure try http://www.dilbert.com/strip/1989-04-16 if the webpage is still alive.''']
+			xbmcaddon.Addon().log(msg, xbmc.LOGERROR)
+			xbmcgui.Dialog().ok('Error: ' + g_AddonName, str(msg[0]))
+			sys.exit(1)
+		match=g_Pattern.search(page)
+		if match:
+			if g_UseCache:
+				write_cache(date,match.group(1))
+			return match.group(1)
+		else:
+			return ""
+
+def create_random_date(starting_date, ending_date):
+	# Creates a random date between a starting and an end date
+    date_delta = ending_date - starting_date
+    random_days = randrange(date_delta.days)
+    return starting_date + datetime.timedelta(days=random_days)
```

Comment:
Sudden four spaces instead of tabs.
I'm pretty sure the interpreter could handle that but still.
Lets keep it consistent :)